pipeline/metadata/flatten.py

```diff
@@ -366,13 +370,52 @@ def _process_satellite_v2(self, scan: Any,
         'error': scan.get('error', None),
         'anomaly': scan['anomaly'],
         'success': not scan['connect_error'],
+        'controls_failed': not scan['passed_control'],
         'received': None,
         'rcode': [],
         'measurement_id': random_measurement_id
     }
     received_ips = scan.get('response')
     yield from self._process_received_ips(row, received_ips)
 
+  def _process_satellite_v2_responses(
+      self, scan: Any, random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Satellite response data.
+
+      Args:
+        scan: a loaded json object containing the parsed content of the line
+        random_measurement_id: a hex id identifying this individual measurement
+
+      Yields:
+        Rows
+    """
+    row = {
+        'domain': scan['test_url'],
+        'category': self.category_matcher.match_url(scan['test_url']),
+        'ip': scan['vp'],
+        'anomaly': None,
+        'success': not scan['connect_error'],
+        'controls_failed': not scan['passed_control'],
+        'measurement_id': random_measurement_id
+    }
+    responses = scan.get('response', [])
+    if responses:
+      row['date'] = responses[0]['start_time'][:10]
+      row['start_time'] = format_timestamp(responses[0]['start_time'])
+      row['end_time'] = format_timestamp(responses[-1]['end_time'])
+      row['rcode'] = [str(resp['rcode']) for resp in responses]
+      row['error'] = ' | '.join(
+          [resp['error'] for resp in responses if resp['error']])
+      for resp in responses:
+        if resp['url'] == row['domain']:
+          # Check response for test domain
+          if resp['rcode'] == 0 and resp['has_type_a']:
+            # Valid answers
+            row['has_type_a'] = True
+            for ip in resp['response']:
```

Comment:
If there are multiple ips in `resp['response']` won't this just add the first and yield immediately instead of iterating over all of them?

Reply:
The flattened rows for Satellite each have one answer IP for tagging purposes (this is similar to `_process_received_ips`).

Reply:
In that case I would just write it as

```
ip = response['response'][0]
```

so the intent is clearer

Reply:
Just to clarify, `response['response']` can have multiple ips, but this function is flattening the measurement so that each yielded row has one answer IP. 

`test_flatten.py` has an example of this in `test_flattenmeasurement_dns`. The `response['response']` field has three answer IPs, resulting in three output rows.
```
{
          "vp": "8.8.4.4",
          "test_url": "login.live.com",
          "response": [
            ...
            {
              "url": "login.live.com",
              "has_type_a": true,
              "response": [
                "40.126.31.135",
                "40.126.31.8",
                "40.126.31.6"
              ],
              ...
            },
            ...
          ],
          ...
}
```